mmtrack/datasets/sot_dataset.py

```diff
@@ -0,0 +1,227 @@
+import os.path as osp
+import random
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+from mmcv.utils import print_log
+from mmdet.datasets.pipelines import Compose
+from torch.utils.data import Dataset
+
+from mmtrack.core.evaluation import eval_sot_ope
+from mmtrack.datasets import DATASETS
+
+
+@DATASETS.register_module()
+class SOTDataset(Dataset, metaclass=ABCMeta):
+    """Dataset of single object tracking.
+
+    The dataset can both support training and testing mode.
+    """
+
+    CLASSES = None
+
+    def __init__(self, img_prefix, pipeline, split, test_mode=False, **kwargs):
+        self.img_prefix = img_prefix
+        self.split = split
+        self.pipeline = Compose(pipeline)
+        self.test_mode = test_mode
+        self.is_video_dataset = True
+        self.bbox_min_size = 0
+        ''' The self.data_info is a list, which the length is the
+            number of videos. The default content is in the following format:
+            [
+                {
+                    'video_path': the video path
+                    'ann_path': the annotation path
+                    'start_frame': the starting frame number contained in the
+                                    image name
+                    'end_frame': the ending frame number contained in the image
+                                    name
+                    'num_digit': the number of digits in the image name
+                },
+                ...
+            ]
+        '''
+        self.data_infos = self.load_data_infos(split=self.split)
+        self.num_frames_per_video = []
+        for video_ind in range(len(self.data_infos)):
+            self.num_frames_per_video.append(self.get_len_per_video(video_ind))
```

Comment:
`self.num_frames_per_video = [self.get_len_per_video(video_ind) for video_ind in range(len(self.data_infos))]`